File: Half Pyramid Using Numbers/halfNumberPyramid.java
import java.util.Scanner;

class halfNumberPyramid {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    System.out.println("Enter the value of n : ");
    int n = sc.nextInt();

    for (int i=1; i<=n; i++) {
      for (int j=1; j<=n; j++) {
        if (j <= i) {
          System.out.print(i);
        }
        else {
          System.out.print(" ");
        }
      }
      System.out.print("\n");
    }

    sc.close();
  }
}

File: Half Pyramid Using Numbers/halfNumberPyramid.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

// Function to print star pattern.
function halfNumberPyramid(n) {
  for (let i=1; i<=n; i++) {
    let pattern = "";
    for (let j=1; j<=n; j++) {
      if (j <= i) {
        pattern += i;
      }
      else {
        pattern += " ";
      }
    }
    console.log(pattern);
  }
}

// value of n from user and calling halfNumberPyramid on them.
rl.question('Enter the value of n : ', n => {
  halfNumberPyramid(Number(n));
  rl.close();
});

